main.go

```diff
@@ -57,7 +57,13 @@ func init() {
 func printVersion() {
 	setupLog.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
 	setupLog.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
-	setupLog.Info(fmt.Sprintf("baremetal-operator version: %s", version.String))
+	if len(version.Commit) > 0 {
```

Comment:
Maybe we can default those strings to something like "unset" so we always have something to print, and it's obvious when a value wasn't provided.

Reply:
Sure, would UNSET in caps work? Capitals are for a user to notice that those values where not defined.

Reply:
Rebased with `unknown`, looks better in my opinion.